src/debuggerCommunicationServer.ts

```diff
@@ -106,6 +108,11 @@ export class DebuggerCommunicationServer {
                     });
                 }
             });
+            socket.on("disconnect_confirmation", () => {
+                this.serverIo.close();
+                this.serverHttp.close();
+                console.info("Closing the server");
+            });
```

Comment:
Restarting works pretty smooth for me, but stopping the debugger and starting it again gives me the error message: 
```
Warning : The Debugger Server cannot be opened. Please try to free the port 5577 if it's already in use or select another one in your Settings 'Device Simulator Express: Debugger Server Port' and start another debug session. You can still debug your code but you won't be able to use the Simulator.
```
Everything seems to work, though?

Reply:
I wonder if we are losing the disconnect event by the time we start it again?

Reply:
Yea, after checking. Since we disconnects the python socket io server at the same time we emit. We cannot intercept the confirmation. My solution is to disconnect the front end with a timeout once the message is sent to disconnect the python side

Reply:
@nasadigital @isadorasophia it should be fixed :) . You were right, the message was lost, so the socket io on the front end never disconnected. Using a timeout works better.

Reply:
@xnkevinnguyen I think restarting is broken for me now.
Below is a stack trace:
```
Traceback (most recent call last):
  File "c:\Users\daseramo\.vscode\extensions\ms-python.python-2020.2.64397\pythonFiles\ptvsd_launcher.py", line 48, in <module>
    main(ptvsdArgs)
  File "c:\Users\daseramo\.vscode\extensions\ms-python.python-2020.2.64397\pythonFiles\lib\python\old_ptvsd\ptvsd\__main__.py", line 432, in main        
    run()
  File "c:\Users\daseramo\.vscode\extensions\ms-python.python-2020.2.64397\pythonFiles\lib\python\old_ptvsd\ptvsd\__main__.py", line 316, in run_file    
    runpy.run_path(target, run_name='__main__')
  File "C:\Users\daseramo\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "C:\Users\daseramo\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "C:\Users\daseramo\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "c:\Users\daseramo\.vscode\extensions\vscode-python-devicesimulator\out\debug_user_code.py", line 43, in <module>
    debugger_communication_client.init_connection(server_port)
  File "c:\Users\daseramo\.vscode\extensions\vscode-python-devicesimulator\out\common\debugger_communication_client.py", line 46, in init_connection     
    sio.connect("http://localhost:{}".format(port))
  File "c:\Users\daseramo\.vscode\extensions\vscode-python-devicesimulator\out\python_libs\socketio\client.py", line 264, in connect
    six.raise_from(exceptions.ConnectionError(exc.args[0]), None)
  File "<string>", line 3, in raise_from
socketio.exceptions.ConnectionError: Connection refused by the server
```

Reply:
Should be completely fixed with the latest changes :) . I have configured different flow for restart and stop debugging session as they don't have the same behaviour. Restart will not disconnect the socket io server, but stop will disconnect.

Reply:
I am getting same error as previously on stop -> start:
```Warning : The Debugger Server cannot be opened. Please try to free the port 5577 if it's already in use or select another one in your Settings 'Device Simulator Express: Debugger Server Port' and start another debug session. You can still debug your code but you won't be able to use the Simulator.```
And debugger seems to work.